### DISCUSSION TITLE
Current Benchmarks

### DISCUSSION THREAD

**Discussion by whiterabbit963** [taocpp/json#82] [Q&A]
Have benchmarks been recently tested? I am using the current master and I have a situation where I am parsing a 1MB json stream (third party data). In Debug mode it takes over 6 seconds to read into a tao::json::value. In Release mode this is significantly better, taking around 0.05 seconds. I ran some tests against nlohmann for a comparison. In Debug I attain 0.7 seconds, and in Release 0.02 seconds.
For reference, I used
```
tao::json::value obj = tao::json::from_file(path);
```
and
```
ifstream ifs(path);
nlohmann::json obj;
ifs >> obj;
```

**Comment by ColinH** [comment]
JSON benchmarks are not something we looked into recently. I'm not surprised by debug begin so slow as we are sometimes quite generous with `assert()`, is there any other difference between your debug and release builds? Being slower than NLohmann/JSON is a bit surprising as we usually were much faster, on the other hand we haven't been improving much lately. We'll probably revisit this subject after the PEGTL 3.0 release which might be RSN.

**Comment by whiterabbit963** [comment]
When I get a chance later today, I want to run another test where I dump the file data to a string, and parse that instead. The slowdown might be a result of piecemeal file-io.

**Comment by whiterabbit963** [comment]
I ran some additional tests. ```tao::json::from_string()``` and ```tao::json::from_file()``` execute in similar time. Running with a debugger attached took 10 seconds. From within the gui application, this was taking the previously mentioned 6 second. From a simple command line application, this became 10 seconds. Running the debug code without an attached debugger was significantly faster and took 1 second.

This was all from the latest version of MSVC 2019. Some quick testing of 2017 seemed to produce similar results.

**Comment by ColinH** [comment]
We always build with optimizations enabled as our library is heavily dependent on them. Debug builds are not really supported as we never use them. The only distinction that we might make is whether assertions are enabled or not. How do the libraries compare with release builds?
